app/controllers/users/sessions_controller.rb

```diff
@@ -3,18 +3,22 @@
 class Users::SessionsController < Devise::SessionsController
   before_action :redirect_to_dashboard, only: %i[sign_in_with_token redirect_from_magic_link]
   before_action :ensure_login_token_valid, only: %i[sign_in_with_token redirect_from_magic_link]
+  TEST_USERS = %w[admin@example.com].freeze
 
   def create
     self.resource = validate_email
+
     if resource.errors.any?
-      render :new
+      render :new and return
```

Comment:
What does the 'and return' add to this? i think render was returning that view or does return mean something else in this instance?

Reply:
I added an extra 'branch' to our logic - `user_requires_magic_link`.

And it became apparent that the whole method will read nicer if we use an early return.